src/fixed_rational.jl

```diff
@@ -43,6 +43,8 @@ Base.convert(::Type{Rational{R}}, x::F) where {R,F<:FixedRational} = Rational{R}
 Base.convert(::Type{Rational}, x::F) where {F<:FixedRational} = Rational{eltype(F)}(x.num, denom(F))
 Base.convert(::Type{AF}, x::F) where {AF<:AbstractFloat,F<:FixedRational} = convert(AF, x.num) / convert(AF, denom(F))
 Base.round(::Type{T}, x::F) where {T,F<:FixedRational} = div(convert(T, x.num), convert(T, denom(F)), RoundNearest)
+Base.promote(x::Integer, y::F) where {F<:FixedRational} = (F(x), y)
+Base.promote(x::F, y::Integer) where {F<:FixedRational} = (x, F(y))
 Base.promote(x, y::F) where {F<:FixedRational} = promote(x, convert(Rational, y))
 Base.promote(x::F, y) where {F<:FixedRational} = promote(convert(Rational, x), y)
```

Comment:
Aside, but I think the recommended way to do this is to overload promote_rule. There is some logic to try combinations for you.
```
help?> Base.promote_rule
  promote_rule(type1, type2)

  Specifies what type should be used by promote when given values of types type1 and type2. This
  function should not be called directly, but should have definitions added to it for new types as
  appropriate.

julia> methods(Base.promote)
# 6 methods for generic function "promote" from Base:
 [1] promote()
     @ promotion.jl:388
 [2] promote(x::T, y::T, zs::T...) where T
     @ promotion.jl:409
 [3] promote(x, y, z)
     @ promotion.jl:397
 [4] promote(x, y, z, a...)
     @ promotion.jl:403
 [5] promote(x, y)
     @ promotion.jl:391
 [6] promote(x)
     @ promotion.jl:389

julia> methods(Base.promote_rule)
# 191 methods for generic function "promote_rule" from Base:
...
   [9] promote_rule(::Type{Float64}, ::Type{Bool})
     @ float.jl:166
  [10] promote_rule(::Type{Float64}, ::Type{UInt32})
     @ float.jl:166
  [11] promote_rule(::Type{Float64}, ::Type{Int16})
     @ float.jl:160
  [12] promote_rule(::Type{Float64}, ::Type{Int8})
     @ float.jl:160
  [13] promote_rule(::Type{Float64}, ::Type{UInt16})
     @ float.jl:166
  [14] promote_rule(::Type{Float64}, ::Type{UInt8})
     @ float.jl:166
  [15] promote_rule(::Type{Float64}, ::Type{Int32})
     @ float.jl:160
  [16] promote_rule(::Type{Float64}, ::Type{UInt128})
     @ float.jl:173
  [17] promote_rule(::Type{Float64}, ::Type{Float16})
     @ float.jl:400
  [18] promote_rule(::Type{Float64}, ::Type{UInt64})
     @ float.jl:166
  [19] promote_rule(::Type{Float64}, ::Type{Int128})
     @ float.jl:174
  [20] promote_rule(::Type{Float64}, ::Type{Float32})
     @ float.jl:401
...
```

Reply:
Thanks, I didn’t realize this. I guess the only advantage of overloading `promote` instead is I could do things like calling `rationalize` rather than just `convert(Rational` (which does a fairly bad job — approximating floats for whatever reason).

Reply:
the problem is it does a very good job converting floats to rationals exactly.

Reply:
see https://github.com/JuliaLang/julia/pull/43427 which will make it so `rationalize` works for numbers that aren't abstractfloats.

Reply:
It's weird that that is the default behavior:

```julia
julia> convert(Rational, 0.2)
3602879701896397//18014398509481984

julia> rationalize(0.2)
1//5
```

but both fractions evaluate to 0.2:

```julia
julia> 3602879701896397/18014398509481984
0.2

julia> 1 / 5
0.2
```

Reply:
Okay I think I will just stick with `Float64` for now, and be explicit about it in the docstring.